src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -59,6 +65,57 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _post_or_schedule_by_key(self, request: Request):
+        request.relay_request_data.get("publicKeys")
+        public_keys = set(request.relay_request_data.get("publicKeys") or ())
+
+        configs = {}
+        pending = []
+        for key in public_keys:
+            computed = self._compute_proj_config(key)
+            if not computed:
+                pending.append(key)
+            else:
+                configs[key] = computed
+
+        if len(pending) > 0:
+            configs["pending"] = pending
+
+        return Response(configs, status=200)
+
+    def _compute_proj_config(self, public_key):
+        """
+        Returns the config of a project if it's in the cache; else, schedules a
+        task to compute and write it into the cache, and returns False.
+
+        Debouncing of the project happens after the task has been scheduled.
+        """
+        cached_config = projectconfig_cache.get(public_key)
+        if cached_config:
+            return cached_config
+        if projectconfig_debounce_cache.is_debounced(
```

Comment:
Rather than calling low-level functions that need to know about the internals of the project config cache here, could we create a more semantic function like `get_or_schedule` that internally does these steps. Internally, that function should in-turn call `schedule_update_config_cache` to maximize reuse of existing logic.

IMHO, the public interface of the projectconfig cache should not have methods like `is_debounced` at all.